pep-0580.rst

```diff
@@ -122,39 +122,36 @@ Parent
 ------
 
 The ``cc_parent`` field (accessed for example by a ``__parent__``
-or ``__objclass__`` descriptor from Python code) can be any Python object.
-For methods of extension types, this is set to the class.
-For functions of modules, this is set to the module.
-However, custom classes are free to set ``cc_parent`` to whatever they want,
-it can also be ``NULL``.
-It is only used by the C call protocol if the ``CCALL_OBJCLASS`` flag is set.
+or ``__objclass__`` descriptor from Python code) can be any Python
+object, or NULL.
+Custom classes are free to set ``cc_parent`` to whatever they want.
+It is only used by the C call protocol if the
+``CCALL_OBJCLASS`` flag is set.
 
-The parent serves multiple purposes: for methods of extension types
-(more precisely, when the flag ``CCALL_OBJCLASS`` is set),
-it is used for type checks like the following::
+For methods of extension types, ``cc_parent`` points to the class
+that defines the method (which may be a superclass of ``type(self)``).
+This is currently is non-trivial to retreive from a method's code.
+(In the future, this can be used to access to the module state via
+the defining class. See the rationale of PEP 573 for details.)
+
```

Comment:
It would be good to mention here the value of `cc_parent` for functions of a module. It looks strange to me to wait with that until the end of this section.

Reply:
I think it's good to complete the "For methods of extension types" discussion with the type check, before diving into "For functions of modules".

Reply:
> I think it's good to complete the "For methods of extension types" discussion with the type check

On the other hand, the parts about "methods of extension types" and "functions of a module" both describe the mapping of current functionality onto the new protocol. The part about `CCALL_OBJCLASS` is describing the protocol itself. It feels strange to me to mix that.

Maybe a good solution would be to put the last part about functions of module first, so we get the order:
- functions of modules
- methods of extension types
- `CCALL_OBJCLASS`